File: Two Sum/two sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        vector<int> ans;
        
        for(int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            
            if (mp.find(complement) != mp.end()) {
                ans.push_back(mp[complement]);
                ans.push_back(i);
                break;
            }
            mp[nums[i]] = i;
        }
        
        return ans;
    }
};

File: Two Sum/two sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> ht = new HashMap<>();
        int[] ans = new int[2];
        for(int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (ht.containsKey(complement)) {
                ans[0] = ht.get(complement);
                ans[1] = i;
                break;
            }
            ht.put(nums[i], i);
        }
        
        return ans;
    }
}